### PR TITLE
feat(actions): user login action now is ajaxed

### PR DESCRIPTION
**Pull request by jdalsem**
No more page refresh on faulty logins...

### CODE REVIEW

**Comment by jdalsem** [comment]
dont merge yet... something wrong

File: actions/login.php [hunk 1]
```diff
@@ -76,11 +75,16 @@
 	$message = elgg_echo('loginok');
 }
 
+system_message($message);
```
Comment (hypeJunction) [inline actions/login.php]: Ok response accepts a message

File: views/default/forms/login.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+define(['jquery', 'elgg', 'elgg/Ajax'], function($, elgg, Ajax) {
+	var ajax = new Ajax();
+	
+	$(document).on('submit', '.elgg-form-login', function(e) {
+		var $form = $(this);
+		var data = {};
+		$($form.serializeArray()).each(function(index, obj){
```
Comment (mrclay) [inline views/default/forms/login.js]: You can use the helper:
```js
var data =  ajax.objectify($form);
```

**Comment by jdalsem** [comment]
For some reason error_response result in a 200 ok... was that always the case? 

I had it working as i expected.. don't know what changed.. maybe a rebase, but after that it does not work anymore... the ajax calls return JSON, but are not picked up by the generic error handling, meaning the error message are not showing for the user...

**Comment by mrclay** [comment]
In docs for `elgg_error_response`: "For BC reasons and due to the logic in the client-side AJAX API, this defaults to 200"

**Comment by jdalsem** [comment]
That explains the 200, but not why registered errors are not picked up in
JS and showed to the user...

Op 15 nov. 2016 11:04 p.m. schreef "Steve Clay" <notifications@github.com>:

**Comment by mrclay** [comment]
Definitely a problem with error response. It's supposed to return an Ajax2 wrapper.

**Comment by hypeJunction** [comment]
If it's 200, it doesn't treat it as error. Legacy AJAX API does, but not elgg/Ajax. I think we never figured out what to do with those. There are several open tickets in the Http responses milestone.

File: views/default/forms/login.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	var ajax = new Ajax();
+	
+	$(document).on('submit', '.elgg-form-login', function(e) {
+		var $form = $(this);
+		var data = ajax.objectify($form);
+
+		ajax.action($form.attr('action'), {data: data}).done(function(json) {
+			console.log(json);
+			
+			if (json.forward !== undefined) {
```
Comment (hypeJunction) [inline views/default/forms/login.js]: Use `typeof` 
https://scottdowne.wordpress.com/2010/11/28/javascript-typeof-undefined-vs-undefined/

File: views/default/forms/login.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+define(['jquery', 'elgg', 'elgg/Ajax'], function($, elgg, Ajax) {
+	var ajax = new Ajax();
+	
+	$(document).on('submit', '.elgg-form-login', function(e) {
+		var $form = $(this);
+		var data = ajax.objectify($form);
+
+		ajax.action($form.attr('action'), {data: data}).done(function(json) {
```
Comment (hypeJunction) [inline views/default/forms/login.js]: Split into multiple lines

Comment (hypeJunction) [inline views/default/forms/login.js]: Use `prop` to make sure it's an absolute URL

**Review by hypeJunction — DISMISSED** [review]

**Comment by jdalsem** [comment]
With the ajax fix #10601 by Steve all is fine now.

**Review by hypeJunction — APPROVED** [review]

**Pull request merged**